cloud_attack_range.py

```diff
@@ -82,54 +60,149 @@
+    if not simulation_techniques:
+        simulation_techniques = 'no'
+    if not attack_chain_file:
+        attack_chain_file = 'no'
+    if not clean_up:
+        clean_up = 'no'
 
-    controller = TerraformController(config, log)
+    return controller.simulate(simulation_techniques,attack_chain_file,clean_up)
+
+
+def dump(args):
+    controller, _, _ = init(args)
+    controller.dump_attack_data(args.dump_name, args.last_sim)
+
+
+def replay(args):
+    controller, _, _ = init(args)
+    controller.replay_attack_data(args.dump_name, args.dump)
+
+
+def build(args):
+    controller, _, _ = init(args)
+    controller.build()
 
-    if list_machines:
-        controller.list_machines()
-        sys.exit(0)
 
-    if action == 'build':
-        controller.build()
+def destroy(args):
+    controller, _, _ = init(args)
+    controller.destroy()
 
-    if action == 'destroy':
-        controller.destroy()
 
-    if action == 'stop':
-        controller.stop()
+def stop(args):
+    controller, _, _ = init(args)
+    controller.stop()
 
-    if action == 'resume':
-        controller.resume()
 
-    if action == 'simulate':
-        controller.simulate(simulation_technique, simulation_file, force, simulation_vars)
+def resume(args):
+    controller, _, _ = init(args)
+    controller.resume()
 
-    if action == 'dump':
-        controller.dump(dump_name)
 
+def test(args):
+    controller, _, _ = init(args)
+    return controller.test(args.test_file)
 
-# rnfgre rtt ol C4G12VPX
+def main(args):
+    # grab arguments
+    parser = argparse.ArgumentParser(
+        description="Use `cloud_attack_range.py action -h` to get help with any Attack Range action")
+    parser.add_argument("-c", "--config", required=False, default="cloud_attack_range.conf",
+                        help="path to the configuration file of the attack range")
+    parser.add_argument("-v", "--version", default=False, action="version", version="version: {0}".format(VERSION),
+                        help="shows current attack_range version")
+    parser.set_defaults(func=lambda _: parser.print_help())  
+
+    actions_parser = parser.add_subparsers(title="Attack Range actions", dest="action")
+    configure_parser = actions_parser.add_parser("configure", help="configure a new attack range")
+    build_parser = actions_parser.add_parser("build", help="Builds attack range instances")
+    #simulate_parser = actions_parser.add_parser("simulate", help="Simulates attack techniques")
+    destroy_parser = actions_parser.add_parser("destroy", help="destroy attack range instances")
+    stop_parser = actions_parser.add_parser("stop", help="stops attack range instances")
+    resume_parser = actions_parser.add_parser("resume", help="resumes previously stopped attack range instances")
+    show_parser = actions_parser.add_parser("show", help="list machines")
+    # test_parser = actions_parser.add_parser("test")
```

Comment:
I would remove these comment out sections, clutters up the code a bit.

Reply:
will do in a new PR